src/sentry/api/serializers/rest_framework/sentry_app.py

```diff
@@ -174,4 +162,26 @@ def validate(self, attrs):
         if not get_current_value("isInternal") and not get_current_value("author"):
             raise ValidationError({"author": "author required for public integrations"})
 
+        # validate globally unique name for public integrations, and org-level uniqueness for internal integrations
+        current_name = get_current_value("name")
+        current_is_internal = get_current_value("isInternal")
+        if current_name:
+            slug = generate_slug(
```

Comment:
@nolachen I am confused why we are generating a slug here

Reply:
@scefali i'm just reusing the logic from the `sentryapp` model file so that we're not putting the slug logic in two files. Before my changes, we were just calling `slugify(name)` to get the slug but the problem is that for internal integrations, the slug also includes a hash at the end

Reply:
@nolachen so If I am understanding this correctly, this check is to ensure the slug is unique even though we generate the slug separately? That seems very error-prone if someone decides to generate the slug a different way. It also seems we are not checking the uniqueness of the actual name which might be OK.

Reply:
@scefali Yes we're only checking that the slug is unique and not the name (not sure if this is correct behavior but that's how it was before too). And I think this is less error prone than manually putting the logic for generating the slug in this file, since i'm just reusing the function that generates the _actual_ slug in this validator. So if someone changes the generate_slug logic, it will be ok here too since it's using the same function

Reply:
I ended up moving the slug generation logic to the creator so we generate, validate, and store it in the same place.